### PR TITLE
auth2 refactor

### PR DESCRIPTION
**Pull request by rodrigopavezi**
@ezequielc please review.

This was staged on https://editor-stage-4.risevision.com/

cheers

### CODE REVIEW

**Comment by ezequielc** [comment]
@rodrigopavezi I don't see the help drop down. Do you need to rebase?

**Comment by rodrigopavezi** [comment]
@ezequielc good catch, I missed the gulp build. cheers

File: src/js/svc-userstate.js [hunk 1]
```diff
@@ -163,16 +163,16 @@
...
-        //cancel any existing $interval(s)
-        $interval.cancel(_accessTokenRefreshHandler);
-        _accessTokenRefreshHandler = $interval(function () {
-          //cancel current $interval. It will be re-sheduled if authentication succeeds
-          $interval.cancel(_accessTokenRefreshHandler);
-          //refresh Access Token
-          _authorize(true);
-        }, 55 * 60 * 1000); //refresh every 55 minutes
-      };
+      // var _scheduleAccessTokenAutoRefresh = function () {
```
Comment (ezequielc) [inline src/js/svc-userstate.js]: @rodrigopavezi Do we still need auto refresh or can we remove this code? Cheers

Comment (rodrigopavezi) [inline src/js/svc-userstate.js]: @ezequielc yes, I don't think we need that anymore. I will remove it. cheers

**Comment by ezequielc** [comment]
@rodrigopavezi Looks good! :+1:

File: dist/js/common-header.js [hunk 2]
```diff
@@ -3738,47 +3727,45 @@ angular.module("risevision.common.geodata", [])
...
+            _authInstance = auth2.getAuthInstance();
 
-                deferred.reject(authResult.error ||
-                  "failed to authorize user");
-              }
-            });
-          }).then(null, deferred.reject); //gapiLoader
+            if (!attemptImmediate) {
+              return _authInstance.signIn({
+                prompt: "select_account"
```
Comment (rodrigopavezi) [inline dist/js/common-header.js]: @ezequielc I have added this option so the signin show the account chooser as we had before. cheers

Comment (ezequielc) [inline dist/js/common-header.js]: @rodrigopavezi :+1:

**Pull request merged**